### ISSUE TITLE
[Feat] New variable formatting option to include variable name

### ISSUE THREAD

**Issue by alsterg** [grafana/grafana#12741]
I have a use case whereby I would like to expand a variable for lucene syntax, but also attach a namespace (string) in-front of the values. For example:

If the variable 'server' holds the values 'server1' & 'server2' then:

`{server:lucene_ns} => ("server:server1" OR "server:server2")`

As you can see the variable name is used as a prefix before each expanded value, separated with a ':' from the value itself.

I'm submitting a patch that implements this, in the hope that this functionality is useful to other users.

Grafana version: master 0550ec6de1070e9ef90a8618ee7c3366147606f9
Datasource: simple json datasoure (annotations)
OS: Ubuntu 17.10

**Comment by alsterg** [comment]
```
diff --git a/docs/sources/reference/templating.md b/docs/sources/reference/templating.md
index ce1a1299d..6548e9cb3 100644
--- a/docs/sources/reference/templating.md
+++ b/docs/sources/reference/templating.md
@@ -52,6 +52,7 @@ Filter Option | Example | Raw | Interpolated | Description
 `csv`| ${servers:csv} |  `'test1', 'test2'` | `test1,test2` | Formats multi-value variable as a comma-separated string
 `distributed`| ${servers:distributed} | `'test1', 'test2'` | `test1,servers=test2` | Formats multi-value variable in custom format for OpenTSDB.
 `lucene`| ${servers:lucene} | `'test', 'test2'` | `("test" OR "test2")` | Formats multi-value variable as a lucene expression.
+`lucene_ns`| ${servers:lucene_ns} | `'servers:test', 'servers:test2'` | `("servers:test" OR "servers:test2")` | Same as 'lucene' but prefixes the variable name infront of each value separated by ':'.
 
 Test the formatting options on the [Grafana Play site](http://play.grafana.org/d/cJtIfcWiz/template-variable-formatting-options?orgId=1).
 
diff --git a/public/app/features/templating/specs/template_srv.jest.ts b/public/app/features/templating/specs/template_srv.jest.ts
index 86b6aa7ec..e1f2a29cb 100644
--- a/public/app/features/templating/specs/template_srv.jest.ts
+++ b/public/app/features/templating/specs/template_srv.jest.ts
@@ -246,6 +246,11 @@ describe('templateSrv', function() {
       expect(result).toBe('("test" OR "test2")');
     });
 
+    it('multi value and lucene should render as lucene_ns expr', function() {
+      var result = _templateSrv.formatValue(['test1', 'test2'], 'lucene_ns', { name: 'test' });
+      expect(result).toBe('("test:test1" OR "test:test2")');
+    });
+
     it('multi value and regex format should render regex string', function() {
       var result = _templateSrv.formatValue(['test.', 'test2'], 'regex');
       expect(result).toBe('(test\\.|test2)');
diff --git a/public/app/features/templating/template_srv.ts b/public/app/features/templating/template_srv.ts
index fc79d12ff..9104fc3d7 100644
--- a/public/app/features/templating/template_srv.ts
+++ b/public/app/features/templating/template_srv.ts
@@ -70,7 +70,7 @@ export class TemplateSrv {
     return filters;
   }
 
-  luceneFormat(value) {
+  luceneFormat(value, prefix) {
     if (typeof value === 'string') {
       return luceneEscape(value);
     }
@@ -78,7 +78,7 @@ export class TemplateSrv {
```

**Comment by alsterg** [comment]
Sure. I'd like to submit this as a feature request. Please advise

**Comment by marefr** [comment]
Done, i've re-opened, added labels and changed the title

**Comment by alsterg** [comment]
Thanks!

**Comment by hugohaggmark** [comment]
Not sure when this feature was implemented and merged but there is now (in Grafana v7.4.0-pre (db637a3630)) an advanced [formatting option for variables called __`lucene`__](https://grafana.com/docs/grafana/latest/variables/advanced-variable-format-options/#lucene---elasticsearch). I'm going to close this issue now because we're trying to limit open GitHub issues in order to better track planned work and features.

Thank You to you for taking the time to create this issue!
